### PR TITLE
New debug macros

### PR DESCRIPTION
**Pull request by mwu-tow**
### Pull Request Description
This PR introduces new debug macros, as requested in CR of https://github.com/enso-org/ide/pull/1564

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guide.
- [x] All code has been tested where possible.

### CODE REVIEW

File: src/prelude/src/debug/logging.rs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+/// For each given triple `[lower UPPER color]` two symbols are defined:
+/// * a function `$lower` that writes given text to standard output on native targets and to Web
+///   Console on wasm targets.
+/// * a macro `$UPPER` that wraps the above function with `println`-like syntax.
+///
+/// Note: The first argument `$d` must be `$` (dollar sign). It is used to insert dollar sign in the
+/// nested macro.
+macro_rules! define_debug_macros {
+    ($d:tt $([$lower:ident $upper:ident $color:ident])*) => {$(
+        #[doc="Writes given text either to the stdout (non-wasm) or Web Console (wasm)."]
```
Comment (farmaazon) [inline src/prelude/src/debug/logging.rs:27]: Why do we use `#[doc]` instead of `///`?

**Review by farmaazon — COMMENTED** [review]
TBH I have a mixed feelings regarding this change; it duplicate the loggers we already have, giving the small bonus (possibility to log something without logger)

Maybe we could make some DEBUG logger as a static (lazy_static) variable, so everyone could just do `trace!(DEBUG, "Some message");`?

File: src/prelude/src/debug/logging.rs (hunk 1 above)
Comment (mwu-tow) [reply src/prelude/src/debug/logging.rs:27]: No reason anymore actually.
Previously I wanted to generate a unique comment for each instantiation but this quickly became more trouble than it was worth. I'll replace with `///`.

**Comment by mwu-tow** [comment]
> 
> 
> TBH I have a mixed feelings regarding this change; it duplicate the loggers we already have, giving the small bonus (possibility to log something without logger)
> 
> Maybe we could make some DEBUG logger as a static (lazy_static) variable, so everyone could just do `trace!(DEBUG, "Some message");`?

This is not quite possible, because the requirement was to have this macros in the prelude.
Prelude does not depend on logger, logger uses prelude. So these are actually more "primitive" than logger and cannot be implemented on top of it.

File: src/prelude/src/debug/logging.rs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        WARNING!("foo");
+        ERROR!("foo");
+    }
+    #[wasm_bindgen_test]
+    fn wasm_calls() {
+        TRACE!("foo");
+        DEBUG!("foo");
+        INFO!("foo");
+        WARNING!("foo");
+        ERROR!("foo");
```
Comment (wdanilo) [inline src/prelude/src/debug/logging.rs:81]: justm inor thing - I really dont like "foo" in source code - can you change it to "test" ? :D I. know this is just stupid personal preference but if you don't have any other prefs, please do it! :)

**Review by wdanilo — APPROVED** [review]

**Pull request merged**